src/back-end/lib/db.ts

```diff
@@ -691,14 +691,15 @@ export async function hasAttachmentPermission(connection: Connection, session: S
 
   // If file is an attachment on a proposal, and requesting user has access to the proposal, allow
   if (session.user) {
-    const proposals = await connection('cwuProposalAttachments as attachments')
+    const rawProposals = await connection('cwuProposalAttachments as attachments')
       .innerJoin('cwuProposals as proposals', 'proposals.id', '=', 'attachments.proposal')
       .where({ 'attachments.file': id })
       .select<RawCWUProposal[]>('proposals.*');
 
-    if (proposals.length > 0) {
-      for (const proposal of proposals) {
-        if (await hasReadPermissionCWUProposal(connection, session, proposal.opportunity, proposal.id)) {
+    if (rawProposals.length > 0) {
+      for (const rawProposal of rawProposals) {
```

Comment:
You don't need the `if` condition here, if there aren't any rawProposals, the loop won't run.